File: tests/fixtures/conformance/go/sample.go
// Conformance fixture (BACK-422 Tier 1) — Go
package sample

import (
	"errors"
	"fmt"
	"os"
	"strconv" // unused on purpose -- must still be flagged by imports://
	"strings"
)

func validate(order string) (string, error) {
	if order == "" {
		return "", errors.New("empty order")
	}
	return order, nil
}

func processOrder(order string) (string, error) {
	result, err := validate(order)
	if err != nil {
		return "", err
	}
	f, err := os.OpenFile("/tmp/orders.log", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		return "", err
	}
	defer f.Close()
	fmt.Fprintf(f, "%s", result)
	result = strings.ToUpper(result)
	return result, nil
}

func Run(order string) (string, error) {
	return processOrder(order)
}

// BACK-439b/c fixture addition: loop + field write + call effect, added
// standalone (not touching processOrder's line-numbered asserts), same
// precedent as Rust's count_down (BACK-427/430).
type Batch struct {
	Total int
}

func (b *Batch) Run(items []int) {
	for _, item := range items {
		b.Total = b.Total + item
		cache.Set(item)
	}
}

File: tests/fixtures/conformance/cpp/sample.cpp
// Conformance fixture (BACK-422 Tier 1/2) — C++
#include <cstdio>
#include <string>
#include <vector>  // unused on purpose -- must still be flagged by imports://

#define CHECK_OR_RETURN(cond, val) if (!(cond)) return (val)

int validate(int order) {
    if (order == 0) {
        return -1;
    }
    return order;
}

int process_order(int order) {
    int result = validate(order);
    if (result < 0) {
        return -1;
    }
    CHECK_OR_RETURN(result < 1000, -1);
    FILE *f = fopen("/tmp/orders.log", "a");
    if (f == nullptr) {
        return -1;
    }
    fprintf(f, "%d", result);
    fclose(f);
    result = result * 2;
    return result;
}

int run(int order) {
    return process_order(order);
}

// BACK-439b/c fixture addition: loop + field write + call effect, added
// standalone (not touching process_order's line-numbered asserts), same
// precedent as Rust's count_down (BACK-427/430).
class Batch {
public:
    int total = 0;
    void run(std::vector<int>& items) {
        for (int item : items) {
            this->total = this->total + item;
            cache.set(item);
        }
    }
};
